.circleci/config.yml

```diff
@@ -21,7 +21,30 @@ jobs:
     steps:
     - checkout
 
-    # specify any bash command here prefixed with `run: `
-    - run: go get -v -t -d ./...
-    - run: go install github.com/onsi/ginkgo/ginkgo
-    - run: ginkgo -r -race -cover -failOnPending -randomizeAllSpecs
\ No newline at end of file
+    - run:
+        name: Get dependencies
+        command: |
+          go get github.com/zenaton/examples-go
+          go get github.com/davecgh/go-spew/spew
+          go get github.com/joho/godotenv
+          go get github.com/twinj/uuid
+          go get github.com/onsi/ginkgo/ginkgo
+          go install github.com/onsi/ginkgo/ginkgo
+          go get github.com/onsi/gomega/...
+
+    - run:
+        name: Run unit tests
+        command: ginkgo -r -race -cover -failOnPending -randomizeAllSpecs
+
+    - run:
+        name: Download the agent
+        command: curl https://install.zenaton.com | sh
+
+    # take the copy out when agent bug is fixed
+    - run:
+        name: Run integration tests
+        command: |
+          go get github.com/zenaton/agent-integration/go
```

Comment:
The `checkout` step already does a git clone (and caches the result behind the scene).
If `go get` only does a `git clone`/`pull` maybe you could set up the `working_dir`of the docker image to the GOPATH

Reply:
Example : https://github.com/zenaton/zenaton-ruby/blob/master/.circleci/config.yml#L4

Reply:
@idabmat But doesn't the checkout only git clone the zenaton-go library? Here i'm doing a go-get for the agent-integration repo. I'm a bit confused.

Reply:
My bad I thought this was the agent-integration repo :man_facepalming:

Reply:
ok :). no problem I'll go ahead with the merge then.